src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -37,7 +40,35 @@ public record CollectionSettings(
     String modelName,
     IndexingConfig indexingConfig) {
 
-  public record IndexingConfig(Set<String> allowed, Set<String> denied) {
+  private static final CollectionSettings EMPTY =
+      new CollectionSettings("", false, 0, null, null, null, null);
+
+  public static CollectionSettings empty() {
+    return EMPTY;
+  }
+
+  public DocumentProjector indexingProjector() {
+    // IndexingConfig null if no indexing definitions: default, index all:
+    if (indexingConfig == null) {
+      return DocumentProjector.identityProjector();
+    }
+    // otherwise get lazily initialized indexing projector from config
+    return indexingConfig.indexingProjector();
+  }
+
+  public record IndexingConfig(
+      Set<String> allowed, Set<String> denied, Supplier<DocumentProjector> indexedProject) {
+    public IndexingConfig(Set<String> allowed, Set<String> denied) {
+      this(
+          allowed,
+          denied,
+          Suppliers.memoize(() -> DocumentProjector.createForIndexing(allowed, denied)));
```

Comment:
This is done so that `DocumentProjector` is only created first time it is needed (if at all).